schemafy_lib/src/generator.rs

```diff
@@ -22,7 +22,9 @@ pub struct Generator<'a, 'b> {
     /// the default should be fine.
     pub schemafy_path: &'a str,
     /// The JSON schema file to read
-    pub input_file: &'b Path,
+    pub input_file: Option<&'b Path>,
+    /// The JSON string to interpret, mutually exclusive with `input_file`
+    pub input_json: Option<String>,
```

Comment:
doesn't really matter, but does this need to be an owned string? If you take the previous comment could this not just share the lifetime `'b`? You are just parsing this and never keeping the owned value in memory from `generate_with_schema`?

Reply:
Indeed. One thing I could not see a clean way to solve is this: https://github.com/near/schemafy/pull/1/commits/18fa1800e93e5d853d1662ce1af7526719df1f59#diff-bea88ef3ad765bc1058b75944aab0ac87eb4cd96906670cf16bc2bb45a0eb537R56. I have to clone the string there to unify the arms' types since the first one cannot be borrowed out of the arm. Any ideas?

Reply:
oops, didn't see this comment before review. https://github.com/near/schemafy/pull/1#discussion_r920178899 I suggest just deserializing within the match